src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -564,32 +548,56 @@ public HttpClientChannelAsyncRequest(HttpClientRequestChannel channel)
                         if (_channel.State != CommunicationState.Opened)
                         {
                             // if we were aborted while getting our request or doing correlation, 
-                            // we need to abort the web request and bail
+                            // we need to abort the request and bail
                             Cleanup();
                             _channel.ThrowIfDisposedOrNotOpen();
                         }
 
-                        HttpOutput httpOutput = HttpOutput.CreateHttpOutput(_httpRequestMessage, _factory, request, _factory.IsChannelBindingSupportEnabled, _factory.GetHttpClient(), _factory._authenticationScheme, _timeoutHelper);
+                        bool suppressEntityBody = PrepareMessageHeaders(request);
 
+                        if (!suppressEntityBody)
+                        {
+                            _httpRequestMessage.Content = MessageContent.Create(_factory, request, _timeoutHelper);
+                        }
+
+                        await SendPreauthenticationHeadRequestIfNeeded();
                         bool success = false;
```

Comment:
I just discovered yesterday via an internal email thread (not related to WCF but still using HttpClient), some web services will fail a request for a HEAD request but accept the POST request. This means we shouldn't abort on a failure to issue a HEAD request as the real request might still succeed. I'm going to wrap it in a try/empty catch block with a comment explaining why.